controllers/providers/pg/provider.go

```diff
@@ -111,6 +111,9 @@ func (p *Provider) Status(ctx context.Context) (everestv1alpha1.DatabaseClusterS
 
 	status := p.DB.Status
 	status.Status = everestv1alpha1.AppState(pg.Status.State)
+	if !p.DB.DeletionTimestamp.IsZero() {
+		status.Status = everestv1alpha1.AppStateDeleting
+	}
```

Comment:
Does this code ever get executed?

Based on my understand, in the `reconcileDB()` function in the outer controller, we do an early return if the deletionTimestamp is not zero. Which means `Status()` doesn't get called

Reply:
In this PR the Status() is called separately in the cleanup (like this https://github.com/percona/everest-operator/blob/977c8db0e5b95d8b93021f179be3a5a42c31a7a0/controllers/providers/pg/provider.go#L143-L146) and then the status got applied inside `HandleDBDeletion`.

But let me rethink the whole approach according to your feedback regarding finalizers. Thank you!